### PR TITLE
Remove explicit import pass pipelines from compiler API.

### PR DESCRIPTION
**Pull request by ScottTodd**
These were set up to grant standalone access to the import pipelines "to facilitate integration with frontend workflows", but plumbing that style of explicit pipeline construction across the C and Python APIs is making it more difficult to configure input dialects as optional. The functionality still exists via the "--iree-input-type=" flag, but it must now be used as part of the full "VM pass pipeline".

Progress on https://github.com/iree-org/iree/issues/9302, refactoring will make https://github.com/iree-org/iree/pull/9474 easier.

### CODE REVIEW

File: compiler/src/iree/compiler/API/Compiler.h [hunk 1]
```diff
@@ -53,26 +53,6 @@ MLIR_CAPI_EXPORTED void ireeCompilerOptionsGetFlags(
...
-// requires that XLA control flow has been legalized to SCF or CFG and that
-// no tuples are in the input program.
-MLIR_CAPI_EXPORTED void ireeCompilerBuildMHLOImportPassPipeline(
-    MlirOpPassManager passManager);
-
-// Builds a pass pipeline to lower IREE-compatible TOSA function and ops to
-// be a legal input to IREE. This performs that standalone work that
-// ireeCompilerOptionsSetInputDialectTOSA will do as a one-shot.
-MLIR_CAPI_EXPORTED void ireeCompilerBuildTOSAImportPassPipeline(
-    MlirOpPassManager passManager);
```
Comment (ScottTodd) [inline compiler/src/iree/compiler/API/Compiler.h:74]: If we really want to keep the standalone import pass pipelines, we could add 

```c
ireeCompilerBuildImportPassPipeline(InputDialectOptions inputDialectOptions, MlirOpPassManager passManager)
```

with a trimmed down version of https://github.com/iree-org/iree/blob/ca258bcd0dea4d22b4dddc9ef05382d6bd61afd3/compiler/src/iree/compiler/API/Compiler.cpp#L145-L157

calling a new function in https://github.com/iree-org/iree/blob/ca258bcd0dea4d22b4dddc9ef05382d6bd61afd3/compiler/src/iree/compiler/Pipelines/Pipelines.h#L31-L39

then associated pybind definitions.

WDYT, @stellaraccident / @benvanik?

Comment (benvanik) [reply compiler/src/iree/compiler/API/Compiler.h:74]: Not sure if anyone was using it so can't say whether it was worth it - maybe it was useful for getting post-tosa/mhlo inputs for reproducers? If no immediate uses seems fine to drop for now (smaller surface area).

Comment (stellaraccident) [reply compiler/src/iree/compiler/API/Compiler.h:74]: I'm fine with it. This predated the flags plumb through and everything, so once upon a time, this was the only way to get at this.

**Comment by ScottTodd** [comment]
Ping?

**Review by benvanik — APPROVED** [review]

**Pull request merged**